UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/MetaDataLayer.cs

```diff
@@ -522,7 +580,9 @@ private void DistributeExcessToNearbyCells(ReagentMix excess, Vector3Int origin)
 				if (excess.Total <= 0) return;
 
 				// Skip if the neighbor is not passable
-				if (matrix.IsWallAt(neighbor, true)) continue;
+				if (matrix.IsWallAt(neighbor, true)
+				    || matrix.IsWindowAt(neighbor, true)
+				    || matrix.GetNoGC(neighbor, true, new List<DoorMasterController>()).Any()) continue;
```

Comment:
neighbor might have is Passable precalculated

Reply:
public bool IsOccupied => Type == NodeType.Occupied;

Reply:
tried that on the older PR, didn't work